app/models/host.rb

```diff
@@ -88,11 +88,17 @@ class Host < ApplicationRecord
     left_outer_joins(:test_result_profiles)
   }
 
-  scope :with_failed_rules_count, lambda {
-    sq = Host.left_outer_joins(test_result_profiles: :rule_results)
-             .where(rule_results: { result: RuleResult::FAILED })
+  scope :with_failed_rules_count, lambda { |profile = nil|
+    profile ||= RequestStore.store['scoped_search_context_profiles']
+    profile_ids = profile&.pluck(:id) || []
+
+    sq = Host.left_outer_joins(test_results: :rule_results)
```

Comment:
Only joining with `test_results` as profiles are not necessary for us, we already have their IDs.